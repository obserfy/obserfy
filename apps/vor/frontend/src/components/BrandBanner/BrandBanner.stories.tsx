import { FC } from "react"
import BrandBanner from "./BrandBanner"

export default {
  title: "Core/BrandBanner",
  component: BrandBanner,
  parameters: {
    componentSubtitle: "Just a simple BrandBanner",
  },
}

export const Basic: FC = () => <BrandBanner />
