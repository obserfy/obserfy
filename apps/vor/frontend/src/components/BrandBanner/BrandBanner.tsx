import { StaticImage } from "gatsby-plugin-image"
import { FC } from "react"
import { Flex, Box } from "theme-ui"
import { Trans } from "@lingui/macro"
import { Typography } from "../Typography/Typography"

export const BrandBanner: FC = () => (
  <Box mx="auto" p={3} sx={{ width: "100%", maxWidth: "maxWidth.xsm" }}>
    <a href="https://obserfy.com" sx={{ display: "inline-block" }}>
      <Flex sx={{ alignItems: "center" }}>
        <StaticImage
          src="../../images/logo-standalone.png"
          width={32}
          alt="obserfy logo"
          sx={{ flexShrink: 0 }}
          placeholder="blurred"
        />
        <Typography.Body
          ml={2}
          sx={{ fontSize: 3, fontWeight: "bold", lineHeight: 1.2 }}
        >
          Obserfy{" "}
          <span sx={{ fontWeight: "normal", whiteSpace: "nowrap" }}>
            <Trans>for Teachers</Trans>
          </span>
        </Typography.Body>
      </Flex>
    </a>
  </Box>
)

export default BrandBanner
