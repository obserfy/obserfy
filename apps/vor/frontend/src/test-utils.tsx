/* eslint-disable import/no-extraneous-dependencies */
import { FC, ReactElement } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import { ThemeProvider, Box, Theme } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import { I18nProvider } from "@lingui/react"
import { i18n } from "@lingui/core"
import { en } from "make-plural/plurals"
import theme from "./gatsby-plugin-theme-ui"
// @ts-ignore
import enCatalog from "../i18n/lingui/en/messages.js"

// any used here because the catalogs are generated b lingui
// not our responsibility
i18n.loadLocaleData("en", { plurals: en })
i18n.load("en", enCatalog.messages as any)
i18n.activate("en")

const queryClient = new QueryClient()
const history = createHistory(createMemorySource("/"))

const AllProviders: FC = ({ children }) => {
  return (
    <I18nProvider i18n={i18n}>
      <QueryClientProvider client={queryClient}>
        <LocationProvider history={history}>
          <ThemeProvider theme={theme as Theme}>
            <Box sx={{ fontSize: [16, 20] }}>{children}</Box>
          </ThemeProvider>
        </LocationProvider>
      </QueryClientProvider>
    </I18nProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
