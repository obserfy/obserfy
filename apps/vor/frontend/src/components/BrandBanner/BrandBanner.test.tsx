import { render } from "../../test-utils"
import { Basic } from "./BrandBanner.stories"

describe("BrandBanner", () => {
  it("should render correctly", () => {
    const { container } = render(<Basic />)
    expect(container).toMatchSnapshot()
  })
})
