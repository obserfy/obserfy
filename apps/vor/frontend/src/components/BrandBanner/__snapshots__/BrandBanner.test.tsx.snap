// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrandBanner should render correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: auto;
  margin-right: auto;
  padding: 16px;
  width: 100%;
  max-width: 500px;
}

.emotion-2 {
  display: inline-block;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  margin-left: 8px;
  font-size: 20px;
  font-weight: bold;
  line-height: 1.2;
}

.emotion-5 {
  font-weight: normal;
  white-space: nowrap;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <a
        class="emotion-2"
        href="https://obserfy.com"
      >
        <div
          class="emotion-3"
        >
          <p
            class="emotion-4"
          >
            Obserfy
             
            <span
              class="emotion-5"
            >
              for Teachers
            </span>
          </p>
        </div>
      </a>
    </div>
  </div>
</div>
`;
